llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -1607,11 +1607,19 @@ ChangeStatus AAPointerInfoFloating::updateImpl(Attributor &A) {
     //
     // The RHS is a reference that may be invalidated by an insertion caused by
     // the LHS. So we ensure that the side-effect of the LHS happens first.
+
+    if (!OffsetInfoMap.contains(Usr)) {
+      auto &PtrOI = OffsetInfoMap[CurPtr];
+      OffsetInfoMap.insert(std::make_pair(Usr, PtrOI));
```

Comment:
Isn't this doing exactly what the comment above says to not do?
I.e. you're using another way of writing `OffsetInfoMap[Usr] = OffsetInfoMap[CurPtr]`.

```suggestion
      auto &UsrOI = OffsetInfoMap[Usr];
      auto &PtrOI = OffsetInfoMap[CurPtr];
      UsrOI = PtrOI;
```

Reply:
Doesn't std::make_pair store by [value](https://stackoverflow.com/questions/63522732/are-the-values-stored-in-stdpair-by-reference-or-by-value) ?